web/api/v1/api.go

```diff
@@ -896,6 +899,17 @@ func (api *API) series(r *http.Request) (result apiFuncResult) {
 	return apiFuncResult{metrics, nil, warnings, closer}
 }
 
+func contextErr(err error) *apiError {
```

Comment:
I see we already have a similar util: [returnAPIError](https://github.com/prometheus/prometheus/blob/f252d8a9d1211a898ffd78323cdf5cb5b96c30ea/web/api/v1/api.go#L615), maybe we can adjust and re-use it.

A regression test for this would be great.

Reply:
Thanks, I have changed to use existing returnAPIError. It covers context cancellation error, we can just use it. 
For tests, I have added test for the case when context is cancelled while iteration. Other regression test cases looks to be covered by the existing [tests](https://github.com/prometheus/prometheus/blob/f252d8a9d1211a898ffd78323cdf5cb5b96c30ea/web/api/v1/api_test.go#L502).